### PR TITLE
Updating (finally) the ContactStructuralMechanicsApplication

### PR DESCRIPTION
**Pull request by loumalouomega**
This merge updates the ContactStructuralMechanicsApplication with all the changes from the last 7-8 months

### CODE REVIEW

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
This commit is simply too large to be reviewed. 
As a minimum i can not accept any change which is not purely self contained in the ContactStructuralMechanicsApplication.
That is, as a first step, ensure that no file is touched in the commit which belongs to the core or to another application

**Comment by loumalouomega** [comment]
I already checked yesterday, it just affects the application. I can send
you a screenshot from meld.

File: applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#BPrinter
+
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/README.md:2]: we agreed that this library will be removed no?

File: applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
+#include "table_printer.h"
+#include <stdexcept>
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.cpp:2]: isn't this to be removed?

File: applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.h [hunk 3]
```diff
@@ -0,0 +1,213 @@
+#ifndef BPRINTER_TABLE_PRINTER_H_
+#define BPRINTER_TABLE_PRINTER_H_
+
+#include <iostream>
+#include <iomanip>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <cmath>
+#if defined(USE_BOOST_KARMA)
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.h:10]: please do not include other boost libraries, particularly not spirit

**Review by RiccardoRossi — APPROVED** [review]
ok, i revised it more carefully, and the changes to other applications are really "no changes" (i don't really understand why they are here though...)

i will admit the commit since it doesn't affect other people. however i think we reached the agreement of removing the table printer dependency. Please do it... (in another PR to simiplify the handling)

File: applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.h (hunk 3 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.h:10]: I will remove it in a future PR, rightly now are not used anyway

File: applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/README.md (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/README.md:2]: Yes, but in a future PR, there is a Pending PR, but pooyan  told me to rework from the scratch for the logging macros

File: applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_external_libraries/bprinter/table_printer.cpp:2]: In another PR

**Pull request merged**